modules/core/src/main/java/org/apache/ignite/internal/commandline/CommonArgParser.java

```diff
@@ -310,20 +333,36 @@ ConnectionAndSslParameters parseAndValidate(Iterator<String> rawArgIter) {
 
                         break;
 
+                    case CMD_USER_ATTR:
+                        userAttrStr = argIter.nextArg("Expected user attribute string.");
+
+                        break;
+
+                    case CMD_USER_ATTR_PATH:
+                        userAttrPath = argIter.nextArg("Expected user attribute file.");
+
+                        break;
+
                     default:
                         throw new IllegalArgumentException("Unexpected argument: " + str);
                 }
             }
         }
 
+        if (userAttrStr != null)
```

Comment:
I suggest turning parse* methods in pure functions returning maps. Then merge 2 maps explicitly. Then it will be obvious what attribute values take precedence (from command line or from properties file).

Reply:
done